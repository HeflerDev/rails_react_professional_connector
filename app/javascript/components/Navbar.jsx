import React from 'react';
import axios from 'axios';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { logoutUser } from '../actions/index';

import Logo from '../../assets/images/logo.png';

const mapStateToProps = state => ({
  isLoggedIn: state.userReducer.isLoggedIn,
  user: state.userReducer.user.user,
});

const select = dispatch => ({
  logoutUser: () => dispatch(logoutUser()),
});

const ConnectedNavbar = ({ history, isLoggedIn, user, logoutUser }) => {
  const handleClick = () => {
    axios.delete('http://localhost:3001/logout', { withCredentials: true })
      .then(() => {
        logoutUser();
        history.push('/');
      })
  };

  const renderUtilities = () => (isLoggedIn ? (
    <>
    <div className="queue center">
      <div className="hello-text">Hello, { user.username }!</div>
    </div>
    <Link to="/profile" className="queue center" >
      <span>Profile</span>
    </Link>
    <button type="button" onClick={handleClick}>
      <span>Logout</span>
    </button>
    </>
  ) : (
    <>
    <Link to="/signup" className="queue center">
      <span>Sign Up!</span>
    </Link>
    <Link to="/login" className="queue center">
      <span>Login</span>
    </Link>
    </>
  ))

  return (
    <nav className="board queue between">
    <div className="company-logo col-12 col-l-3">
      <Link to="/">
        <img src={Logo} alt="" className="logo" />
        <span>Brand Name</span>
      </Link>
    </div>
    <div className="utils-container queue end col-12 col-m-4 col-l-4">
      { renderUtilities() }
    </div>
    </nav>
  )
};

const Navbar = connect(mapStateToProps, select)(ConnectedNavbar);

export default Navbar;
