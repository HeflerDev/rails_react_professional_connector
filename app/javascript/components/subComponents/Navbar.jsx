import React from 'react';
import axios from 'axios';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { logoutUser } from '../../actions/index';

const mapStateToProps = state => ({
  isLoggedIn: state.userReducer.isLoggedIn,
  user: state.userReducer.user.user,
});

const select = dispatch => ({
  logoutUser: () => dispatch(logoutUser()),
});

const ConnectedNavbar = ({ history, isLoggedIn, user, logoutUser }) => {
  const handleClick = () => {
    axios.delete('http://localhost:3001/logout', { withCredentials: true })
      .then(() => {
        logoutUser();
        history.push('/');
      })
  };

  const renderUtilities = () => (isLoggedIn ? (
    <>
    <p>Hello, { user.username } </p>
    <Link to="/profile" >
      <p>Profile</p>
    </Link>
    <button type="button" onClick={handleClick}>
      <p>Logout</p>
    </button>
    </>
  ) : (
    <>
    <Link to="/signup">
      <p>Sign Up!</p>
    </Link>
    <Link to="/login">
      <p>Login</p>
    </Link>
    </>
  ))

  return (
    <nav className="board queue between">
    <div className="company-logo col-12 col-l-3">
      Logo
    </div>
    <div className="utils-container queue end col-12 col-l-4">
      { renderUtilities() }
    </div>
    </nav>
  )
};

const Navbar = connect(mapStateToProps, select)(ConnectedNavbar);

export default Navbar;
