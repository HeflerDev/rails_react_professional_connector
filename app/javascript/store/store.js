import { createStore, combineReducers } from 'redux';
import userReducer from '../reducers/user';

const rootReducer = combineReducers({
  userReducer,
});

const store = createStore(rootReducer);

export default store;
