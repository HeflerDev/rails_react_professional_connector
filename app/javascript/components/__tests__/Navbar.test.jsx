import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from "react-router-dom";
import store from '../../store/store';
import Navbar from '../Navbar';

describe('when rendering navbar', () => {
  test('it renders correctly', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Navbar />
        </Provider>
      </BrowserRouter>
    )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
