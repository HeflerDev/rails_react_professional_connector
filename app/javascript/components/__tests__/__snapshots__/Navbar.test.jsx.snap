// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when rendering navbar it renders correctly 1`] = `
<nav
  className="board queue between"
>
  <div
    className="company-logo col-12 col-m-4 col-l-3"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt=""
        className="logo"
        src=""
      />
      <span>
        Brand Name
      </span>
    </a>
  </div>
  <div
    className="utils-container queue end col-12 col-m-7 col-l-4"
  >
    <a
      className="queue center"
      href="/signup"
      onClick={[Function]}
    >
      <span>
        Sign Up!
      </span>
    </a>
    <a
      className="queue center"
      href="/login"
      onClick={[Function]}
    >
      <span>
        Login
      </span>
    </a>
  </div>
</nav>
`;
